File: src/types/project.ts
export interface Project {
  id: string;
  name: string;
  description: string;
  imageCount: number;
  createdAt: Date;
  updatedAt: Date;
  imagePairs?: ImagePair[];
  images?: Array<string>;
}

export interface ImagePair {
  id: string;
  projectId: string;
  visibleImage: Image;
  infraredImage: Image;
}

export interface Image {
  id: string;
  url: string;
  name: string;
  type: "visible" | "infrared";
  annotations: Annotation[];
}

export interface Annotation {
  id: string;
  bbox: number[]; // [x, y, width, height]
  label: string;
  confidence?: number;
  color?: string;
  imageId?: string;
}

export interface AutoAnnotateResult {
  annotations: Annotation[];
  error?: string;
}

File: backend/models/project.py
from dataclasses import dataclass
from datetime import datetime
from typing import List, Optional

@dataclass
class ProjectTag:
    name: str
    color: str

@dataclass
class Project:
    id: int
    name: str
    description: Optional[str]
    tags: List[ProjectTag]
    created_at: datetime
    updated_at: datetime 